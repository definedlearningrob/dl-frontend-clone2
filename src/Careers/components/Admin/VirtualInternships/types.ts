import { TOpportunity as TBaseOpportunity } from '@dc/resources/types';

export const enum VirtualInternshipStatuses {
  DRAFT = 'DRAFT',
  PUBLISHED = 'PUBLISHED',
}

export type VirtualInternshipLessonAttributes = {
  lessonId: string;
  step: number;
};

export type FormValues = {
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  description: string;
  name: string;
  pathways: {
    label: string;
    value: string;
  }[];
  availableSpots: number;
  creditsOutcomes: string;
  image: Record<string, unknown>;
  tags: {
    label: string;
    value: string;
  }[];
  requiredExperiences: number;
  status: { value: VirtualInternshipStatuses; label: string };
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};

export type TPathway = Pick<TBaseOpportunity['pathways'][number], 'id' | 'name'>;

export type TOpportunity = Pick<
  TBaseOpportunity,
  | 'id'
  | 'name'
  | 'availableSpots'
  | 'creditsOutcomes'
  | 'description'
  | 'imageUrl'
  | 'opportunityType'
  | 'tags'
  | 'salaryInformation'
> & {
  pathways: TPathway[];
};

export type Lesson = {
  id: string;
  step: number;
  name: string;
  imageUrl: string;
  thumbnailUrl: string;
  type: string;
};

export type TVirtualInternship = {
  archivedAt: string | null;
  badges: {
    id: string;
    name: string;
    imageUrl: string;
  }[];
  id: string;
  opportunity: TOpportunity;
  requiredExperiences: number;
  status: VirtualInternshipStatuses;
  calendarLessons: Lesson[];
  experienceOpportunityLessons: Lesson[];
  postExperienceLessons: Lesson[];
  readinessSkillsLessons: Lesson[];
};
