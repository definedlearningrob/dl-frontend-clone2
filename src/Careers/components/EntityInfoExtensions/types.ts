export type TAdminEntityInfoExtensionField = {
  id: string;
  name: string;
};
