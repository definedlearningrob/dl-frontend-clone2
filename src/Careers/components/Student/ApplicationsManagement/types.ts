export type Counselor = {
  email: string;
  firstName: string;
  fullName: string;
  lastName: string;
  name: string;
  username: string;
  uuid: string;
};

export type AvailableTeacher = {
  email: string;
  fullName: string;
  name: string;
  username: string;
  uuid: string;
};

export type CounselorSelectOption = { label: string; value: string };
