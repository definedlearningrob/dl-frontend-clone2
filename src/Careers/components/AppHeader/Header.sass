@import '@shared/stylesheets/config.sass'

.header-desktop
  +flex-container(row, flex-end, center)
  background-color: $color-background
  height: $desktop-header-height
  max-height: $desktop-header-height
  padding: $m-xs $m-base
  position: sticky
  top: 0
  width: 100%
  z-index: $z-highest
  gap: $m-base

  +media-breakpoint-up(xxxl)
    padding: $m-xs $m-md

  &__shared-slide-info
    +flex-container(row, center, flex-start)
    background-color: $color-secondary-200
    color: $color-secondary-500
    font-size: $font-size-xs
    gap: $m-xs
    margin-left: 234px
    padding: $m-xs
    word-wrap: break-word

  &.-hide-header
    display: none

  &__impersonating-label
    display: block
    font-size: $font-size-xs

  &__impersonate-button.button
    margin-left: $m-base

    .button__text
      color: $color-secondary-500

    &:hover
      +box-shadow-300($color-black)

  &__notification-button
    color: $color-font-secondary

    &.-active
      color: $color-primary-500

    &:hover
      color: $color-primary-500

  &__saving-status
    +flex-container(row, space-evenly, center)

    .icon svg
      fill: $color-success-500

    &.-saving
      p
        color: $color-primary-500

      .icon svg
        animation-duration: 2000ms
        animation-iteration-count: infinite
        animation-name: spin
        animation-timing-function: linear
        fill: $color-success-500

    &.-pending > p
      color: $color-primary-500

  &__saved-icon
    color: $color-success-500

  &__user-dropdown-section
    +flex-container(row, flex-start, center)
    color: $color-font-secondary
    cursor: pointer
    gap: $m-xs
    position: relative
    white-space: nowrap

    &__content
      +box-shadow-200($color-font-primary)
      background-color: $color-white
      border-radius: $border-radius-sm
      color: $color-font-primary
      font-weight: $font-weight-medium
      margin: 0
      padding: $m-xxs
      position: absolute
      right: 0
      top: calc(100% + $m-xxs + $m-xs)
      z-index: $z-higher

      li
        align-items: center
        border-radius: $border-radius-xs * 2
        display: flex
        padding: $m-base * 0.5
        width: 100%

        button
          color: $color-font-primary
          height: $m-md
          margin: 0 $m-xs 0 0
          pointer-events: none
          width: $m-md

        &:hover
          background-color: $color-background
          transition: background-color $transition-duration * 0.5 $transition-function

  &__user-dropdown-button
    +flex-container(row, flex-start, center)
    background: none
    border: none
    color: inherit
    cursor: pointer
    font: inherit
    font-size: $font-size-xs
    font-weight: $font-weight-medium
    outline: none
    padding: 0
    white-space: nowrap

    &.-dropdown-active
      color: $color-primary-500
      fill: $color-primary-500

      .icon
        background-color: $color-white
        border-radius: $border-radius-xs

  &__dropdown-icon
    margin-left: $m-xs
    width: $icon-size

  &.-with-impersonate
    background-color: $color-secondary-500

    .header-desktop__user-dropdown-section
      color: $color-white
      cursor: default

    .header-desktop__user-dropdown-button
      cursor: default
      text-align: left

  .announcements-dropdown
    .dropdown-options
      +box-shadow-300($color-black)
      margin: $m-xs 0 $m-xl + $m-xs
      width: $notifications-dropdown-width

      &__dropdown
        background-color: $color-white
        border-left: 2px solid transparent
        height: auto
        margin-bottom: 1px
        transition: all 0.3s ease-in-out

        &:hover
          border-color: $color-primary-500

      .-read
        background-color: $color-white

    .icon-container
      background-color: $color-background
      border-radius: 50%
      height: $m-md + $m-xs
      margin: 0 $m-xs 0 0
      padding: $m-x $m-xs
      width: $m-md + $m-xs

    .container
      height: 100%
      position: relative
      text-align: left
      width: 100%

    .notifications-dropdown
      &__option-body
        +ellipsis-multiline(2)
        color: $color-font-primary
        line-break: anywhere
        line-height: $line-height-lg
        min-height: 70%

        &__author
          margin-right: $m-xxs
          color: $color-primary-500

  .divider
    background-color: $color-border
    border-radius: $border-radius-xs
    height: 40px
    margin: 0 $m-base
    width: 1px

.presentation-builder-dropdown-button
  background-color: $color-white
  border: 1px solid $color-primary-500
  border-radius: $border-radius-xs
  height: 24px
  margin-left: $m-xs
  width: 24px

  &:hover
    background-color: $color-white !important

  svg
    fill: $color-primary-600

//----- media queries --------------------------------------------------------------------------------------------------


@media (display-mode: fullscreen)
  .-presentation-fullscreen-mode
    .header-desktop,
    .back-button-wrapper
      display: none

.notifications-wrapper
  +flex-container(row, false, center)
  margin-right: $m-md
  gap: $m-base
