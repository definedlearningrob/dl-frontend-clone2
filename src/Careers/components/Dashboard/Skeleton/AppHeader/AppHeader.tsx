import SharedAvatar from '@shared/components/Avatar/Avatar';
import SkeletonRectangle from '@shared/components/Skeleton/Rectangle/Rectangle';

function SkeletonAppHeader() {
  const user = {
    firstName: 'Defined',
    lastName: 'Careers',
  };

  return (
    <>
      <div className='flex justify-end h-[theme(variables.headerHeight)] px-base py-xs xxxl:px-md'>
        <div className='flex items-center gap-xs'>
          <SharedAvatar size='40' theme='light' user={user} />
          <div className='grid gap-xxs grid-rows-2'>
            <SkeletonRectangle className='!w-[100px] !p-xxs' color='darker' />
            <SkeletonRectangle className='!w-[100px] !p-xxs' color='darker' />
          </div>
        </div>
      </div>
    </>
  );
}
export default SkeletonAppHeader;
