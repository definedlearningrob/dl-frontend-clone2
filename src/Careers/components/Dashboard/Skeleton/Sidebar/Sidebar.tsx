import { DCLogo } from '@dc/shared/DCLogo/DCLogo';

import SkeletonSidebarFooter from './Footer/Footer';
import SkeletonSidebarItem from './Item/Item';

type Props = {
  itemsCount: number;
};

function SkeletonSidebar({ itemsCount }: Props) {
  const renderItems = () =>
    Array.from({ length: itemsCount }, (_, index) => <SkeletonSidebarItem key={index} />);

  return (
    <div className='skeleton-sidebar'>
      <DCLogo className='skeleton-sidebar__logo-icon' />
      {renderItems()}
      <div className='skeleton-sidebar__divider' />
      <SkeletonSidebarFooter />
    </div>
  );
}

export default SkeletonSidebar;
