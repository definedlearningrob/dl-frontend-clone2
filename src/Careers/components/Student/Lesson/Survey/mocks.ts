import lessonInCourseQuery from '@dc/graphql/student/queries/lessonInCourse';
import { CREATE_CAREER_REVIEW_SURVEY_ATTEMPT } from '@dc/graphql/student/mutations/createCareerReviewSurveyAttempt';
import createCareerReviewSurveyAnswersMutation from '@dc/graphql/student/mutations/createCareerReviewSurveyAnswers';
import currentCoursesQuery from '@dc/graphql/student/queries/currentCourses';
import { SURVEY_QUESTION_TYPE } from '@dc/components/Student/Lesson/Survey/types';

export const createAttemptSpy = jest.fn();
export const lessonInCourseSpy = jest.fn();

export let createCareerReviewSurveyAnswersSpy = jest.fn();
export let currentCoursesSpy = jest.fn();

export const mocks = [
  {
    request: {
      query: lessonInCourseQuery,
      variables: { courseId: '2', lessonId: '111', track: true },
    },
    result: () => {
      lessonInCourseSpy();

      return {
        data: {
          course: {
            careerName: 'Dental Laboratory Technicians',
            content: {
              checkIns: [],
              items: [],
            },
            hasInstitutionsInStudentState: true,
            id: '2',
            lesson: {
              assignments: [],
              attachments: [],
              careerReviewSurvey: {
                version: 1,
                performed: false,
                questions: [
                  {
                    answer: [],
                    id: '1',
                    options: [
                      {
                        option: 'Yes, definitely',
                        step: 1,
                      },
                      {
                        option: 'Definitely not',
                        step: 2,
                      },
                    ],
                    question: 'Can blind people see their dreams?',
                    type: 'single_choice' as SURVEY_QUESTION_TYPE,
                  },
                  {
                    answer: [],
                    id: '2',
                    options: [
                      {
                        option: 'Very Interested',
                        step: 1,
                      },
                      {
                        option: 'Not at all Interested',
                        step: 2,
                      },
                      {
                        option: 'Other',
                        step: 3,
                      },
                    ],
                    question: 'How interested are you in the Pathway this course is a part of?',
                    type: 'multiple_choice' as SURVEY_QUESTION_TYPE,
                  },
                ],
                __typename: 'CareerReviewSurvey',
              },
              checkInGroups: [],
              checkInQuestions: [],
              description: {
                introduction: null,
                goal: null,
                role: null,
                audience: null,
                situation: null,
                __typename: 'LessonDescription',
              },
              hasPresentation: false,
              id: '609',
              imageUrl: 'http://test.com/image.png',
              name: 'Career Review Survey',
              externalPresentations: [],
              researchLinks: [],
              texts: [],
              type: 'career_review_survey',
              videos: [],
              vocabularies: [],
              __typename: 'Lesson',
            },
            lessons: [
              {
                id: '782',
                step: 4,
                careerReviewSurvey: null,
              },
              {
                id: '609',
                step: 5,
                careerReviewSurvey: {
                  performed: false,
                  version: 1,
                  __typename: 'CareerReviewSurvey',
                },
              },
              {
                id: '275',
                step: 1,
                careerReviewSurvey: null,
              },
              {
                id: '331',
                step: 2,
                careerReviewSurvey: null,
              },
              {
                id: '668',
                step: 3,
                careerReviewSurvey: null,
              },
            ],
            name: 'Dental Laboratory Technician',
            progress: {
              submitted: 10,
              total: 10,
            },
            reviewSurvey: {
              questions: [
                {
                  id: '1',
                },
                {
                  id: '2',
                },
                {
                  id: '3',
                },
                {
                  id: '4',
                },
                {
                  id: '5',
                },
                {
                  id: '6',
                },
                {
                  id: '7',
                },
                {
                  id: '8',
                },
                {
                  id: '9',
                },
                {
                  id: '10',
                },
                {
                  id: '11',
                },
              ],
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: CREATE_CAREER_REVIEW_SURVEY_ATTEMPT,
      variables: { input: { contextType: 'COURSE', contextId: '1' } },
    },
    result: () => {
      createAttemptSpy();

      return {
        data: {
          createCareerReviewSurveyAttempt: {
            careerReviewSurveyAttempt: {
              id: '2',
              status: 'IN_PROGRESS',
            },
          },
        },
      };
    },
  },
  {
    request: {
      query: createCareerReviewSurveyAnswersMutation,
      variables: {
        input: {
          contextId: '1',
          contextType: 'COURSE',
          answers: [
            { questionId: '1', answer: ['Yes, definitely'] },
            { questionId: '2', answer: ['Very Interested'] },
          ],
        },
      },
    },
    result: () => {
      createCareerReviewSurveyAnswersSpy();

      return {
        data: {
          createCareerReviewSurveyAnswers: {
            status: 'success',
          },
        },
      };
    },
  },
  {
    request: {
      query: currentCoursesQuery,
    },
    result: () => {
      currentCoursesSpy();

      return {
        data: {
          currentCourses: {
            id: '1',
            imageUrl: '',
            name: '',
            progress: {
              submitted: 2,
              total: 5,
            },
            status: 'success',
            pathway: {
              name: '',
            },
            thumbnailUrl: '',
            type: 'survey',
          },
        },
      };
    },
  },
];
