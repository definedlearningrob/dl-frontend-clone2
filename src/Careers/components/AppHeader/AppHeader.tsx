import { useState, useRef } from 'react';
import cx from 'classnames';
import { useHistory } from 'react-router-dom';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';

import AppHeaderDropdown from '@dc/components/AppHeader/Dropdown/Dropdown';
import useExpandSidebar from '@dc/hooks/useExpandSidebar';
import SharedTempBackNavButton from '@dc/components/shared/TempBackNavButton/TempBackNavButton';
import useUserInfo from '@dc/hooks/useUserInfo';
import notificationsQuery from '@dc/graphql/student/queries/notifications';
import { TRootState } from '@dc/redux/reducers';

import { LanguageSwitcher } from '@shared/components/LanguageSwitcher/LanguageSwitcher';
import { Notifications } from '@shared/components/Notifications';
import SharedAvatar from '@shared/components/Avatar/Avatar';
import SharedButton from '@shared/components/Button/Button';
import SharedIcon from '@shared/components/Icon/Icon';
import { ReactComponent as ArrowUp } from '@shared/svg/chevron_up.svg';
import { ReactComponent as ArrowDown } from '@shared/svg/chevron_down.svg';
import { NOTIFICATION_TYPES } from '@shared/resources/constants';
import { useNavigation } from '@shared/components/Sidebar/useNavigation';
import { callToast } from '@shared/components/Toaster/Toaster';
import { usePresentationState } from '@shared/hooks/usePresentationState';
import { HelpScout } from '@shared/components/HelpScout/HelpScout';
import { APP_TYPES } from '@shared/resources/enums';
import { useLocalize } from '@shared/hooks/useLocalize';

import PresentationItems from './PresentationItems/PresentationItems';

import './Header.sass';

type Props = {
  beforeOnboarding?: boolean;
};

function AppHeader({ beforeOnboarding }: Props) {
  const node = useRef(null);
  const { hideHeader } = useExpandSidebar();
  const { presentationState, presentationDispatch } = usePresentationState();
  const { isExpanded, isHidden, backNavButton } = useNavigation();
  const { user } = useSelector((state: TRootState) => state.session);
  const { userInfo, impersonateStop } = useUserInfo();
  const { selectedLocale, localesToSelect, setLanguage } = useLocalize();
  const [show, setShow] = useState(false);
  const { t } = useTranslation();
  const history = useHistory();

  const isImpersonated = 'isImpersonated' in userInfo && userInfo.isImpersonated;
  const hasRole = 'role' in userInfo;

  const desktopHeaderClasses = cx('header-desktop', {
    '-expanded-sidebar-left': isExpanded,
    '-with-impersonate': isImpersonated,
    '-animation-open': !isExpanded,
    '-animation-close': isExpanded,
    '-hide-navigation': isHidden,
    '-hide-header': hideHeader,
  });

  const userDropdownClasses = cx('header-desktop__user-dropdown-button', {
    '-dropdown-active': show,
  });

  const stopImpersonate = async () => {
    try {
      const studentUuid = userInfo.uuid;
      const redirectPath = 'role' in userInfo ? '/post-secondary' : `/students/${studentUuid}`;
      await impersonateStop();
      history.replace(redirectPath);
    } catch (error) {
      callToast('error', t('common.error', { error }));
    }
  };

  const setShowPresentationSettings = (payload: unknown) =>
    presentationDispatch({ type: 'SET_SHOW_PRESENTATION_SETTINGS', payload });

  const notificationsTheme = isImpersonated ? 'light' : 'default';

  return (
    <>
      <div className={desktopHeaderClasses} data-testid='header-desktop'>
        {backNavButton.show && (
          <SharedTempBackNavButton
            data-testid='shared-temp-button'
            destination={backNavButton.destination}
            presentationDispatch={presentationDispatch}
            presentationState={presentationState}
            setShowPresentationSettings={setShowPresentationSettings}
            text={backNavButton.text}
            onBack={backNavButton.callback}
          />
        )}
        {!beforeOnboarding && user?.type === 'student' && (
          <div className='notifications-wrapper'>
            <Notifications
              query={notificationsQuery}
              theme={notificationsTheme}
              type={NOTIFICATION_TYPES.GENERAL}
            />
            <Notifications
              query={notificationsQuery}
              theme={notificationsTheme}
              type={NOTIFICATION_TYPES.ANNOUNCEMENT}
            />
          </div>
        )}
        {hasRole && (presentationState.taskId || presentationState.slideLibraryId) && (
          <PresentationItems userRole={userInfo.role} />
        )}
        {user?.type === 'user' && <HelpScout app={APP_TYPES.CAREERS} />}
        <LanguageSwitcher
          languages={localesToSelect}
          selectedLanguage={selectedLocale}
          setLanguage={setLanguage}
        />
        <div
          ref={node}
          className='header-desktop__user-dropdown-section'
          data-testid='dropdown-section'
          // Temporary hide dropdown for impersonated since there is only logout option //
          onClick={() => !isImpersonated && setShow(!show)}>
          <SharedAvatar
            className='header-desktop__header-item'
            size='32'
            theme={show ? 'active' : 'light'}
            user={userInfo}
          />
          <button className={userDropdownClasses} data-testid='app-header-dropdown-trigger'>
            <div>
              {isImpersonated && (
                <div className='text-xs' data-testid='impersonated-label'>
                  {t('appHeader.impersonating')}
                </div>
              )}
              <span data-testid='user-first-last-name'>
                {userInfo?.firstName} {userInfo?.lastName}
              </span>
            </div>
            {/* Temporary hide dropdown for impersonated since there is only logout option */}
            {!isImpersonated && (
              <SharedIcon
                className='header-desktop__dropdown-icon'
                icon={show ? <ArrowUp /> : <ArrowDown />}
                size='xs'
              />
            )}
          </button>
          {show && <AppHeaderDropdown refProp={node} setShow={setShow} />}
          {isImpersonated && (
            <SharedButton
              className='ml-base'
              data-testid='stop-impersonating-button'
              size='sm'
              onClick={stopImpersonate}>
              {t('appHeader.exitSession')}
            </SharedButton>
          )}
        </div>
      </div>
    </>
  );
}
export default AppHeader;
