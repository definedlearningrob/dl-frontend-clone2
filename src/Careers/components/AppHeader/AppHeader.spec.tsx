import { MockedProvider, MockedResponse } from '@apollo/client/testing';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import AppHeader from '@dc/components/AppHeader/AppHeader';
import { Roles } from '@dc/resources/enums';
import { UserInfoProvider } from '@dc/hooks/useUserInfo';
import Notifications from '@dc/graphql/student/queries/notifications';
import { ExpandSidebarProvider } from '@dc/hooks/useExpandSidebar';
import { userInfoMock } from '@dc/tests/mocks/userMocks';
import { studentInfoMock } from '@dc/tests/mocks/studentMocks';

import { NavigationContextProvider } from '@shared/components/Sidebar/useNavigation';
import { PresentationStateProvider } from '@shared/hooks/usePresentationState';
import { TNotificationsData } from '@shared/components/Notifications/types';

describe('AppHeader', () => {
  it('should render correctly for teacher', () => {
    const { container } = renderWithRouterAndReduxProvider(
      <MockedProvider mocks={[userInfoMock]}>
        <ExpandSidebarProvider>
          <PresentationStateProvider>
            <NavigationContextProvider>
              <UserInfoProvider
                value={{
                  userInfo: {
                    ...userInfoMock.result.data.userInfo,
                    role: Roles.TEACHER,
                    firstName: 'Test',
                    lastName: 'Man',
                  },
                }}>
                <AppHeader />
              </UserInfoProvider>
            </NavigationContextProvider>
          </PresentationStateProvider>
        </ExpandSidebarProvider>
      </MockedProvider>
    );

    expect(container).toMatchSnapshot();
  });

  it('should render correctly for student', () => {
    const initialState = {
      session: {
        user: {
          type: 'student',
        },
        loginError: {},
      },
    };

    const notificationMock: MockedResponse<TNotificationsData> = {
      request: {
        query: Notifications,
        variables: { page: 1, perPage: 10, scope: 'ALL', type: 'GENERAL' },
      },
      result: {
        data: {
          notifications: { nodesCount: 0, nodes: [], pagesCount: 1 },
        },
      },
    };

    const announcementsMock: MockedResponse<TNotificationsData> = {
      request: {
        query: Notifications,
        variables: { page: 1, perPage: 10, scope: 'ALL', type: 'ANNOUNCEMENT' },
      },
      result: {
        data: {
          notifications: { nodesCount: 0, nodes: [], pagesCount: 1 },
        },
      },
    };

    const { container } = renderWithRouterAndReduxProvider(
      <MockedProvider mocks={[notificationMock, announcementsMock, studentInfoMock]}>
        <ExpandSidebarProvider>
          <PresentationStateProvider>
            <NavigationContextProvider>
              <UserInfoProvider
                value={{
                  userInfo: {
                    ...studentInfoMock.result.data.userInfo,
                    firstName: 'Test',
                    lastName: 'Man',
                  },
                }}>
                <AppHeader />
              </UserInfoProvider>
            </NavigationContextProvider>
          </PresentationStateProvider>
        </ExpandSidebarProvider>
      </MockedProvider>,
      { initialState }
    );

    expect(container).toMatchSnapshot();
  });
});
