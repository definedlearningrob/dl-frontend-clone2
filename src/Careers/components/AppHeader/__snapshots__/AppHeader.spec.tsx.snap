// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader should render correctly for student 1`] = `
<div>
  <div
    class="header-desktop -animation-open"
    data-testid="header-desktop"
  >
    <div
      class="notifications-wrapper"
    >
      <div
        aria-expanded="false"
        aria-haspopup="menu"
        class="trigger"
        data-state="closed"
        data-testid="notifications-dropdown-trigger"
        id="radix-0"
        type="button"
      >
        <button
          class="icon-button -size-sm -square text-font-secondary hover:!bg-transparent hover:text-primary-500"
          data-testid="icon-button"
          type="button"
        >
          <div
            class="icon icon-button__icon -sm-size"
            data-testid="icon"
          >
            <div />
          </div>
        </button>
      </div>
      <div
        aria-expanded="false"
        aria-haspopup="menu"
        class="trigger"
        data-state="closed"
        data-testid="notifications-dropdown-trigger"
        id="radix-1"
        type="button"
      >
        <button
          class="icon-button -size-sm -square text-font-secondary hover:!bg-transparent hover:text-primary-500"
          data-testid="icon-button"
          type="button"
        >
          <div
            class="icon icon-button__icon -sm-size"
            data-testid="icon"
          >
            <div />
          </div>
        </button>
      </div>
    </div>
    <div
      class="header-desktop__user-dropdown-section"
      data-testid="dropdown-section"
    >
      <div
        class="avatar header-desktop__header-item"
        data-testid="avatar-image"
      >
        <div
          class="avatar__image sb-avatar"
          style="display: inline-block; vertical-align: middle; width: 32px; height: 32px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
        >
          <div
            class="avatar__image sb-avatar__text"
            style="width: 32px; height: 32px; line-height: initial; text-align: center; color: rgb(106, 110, 124); background: rgb(252, 252, 252); border-radius: 100%;"
            title="Test Man"
          >
            <div
              style="display: table; table-layout: fixed; width: 100%; height: 100%;"
            >
              <span
                style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
              >
                <span />
              </span>
            </div>
          </div>
        </div>
      </div>
      <button
        class="header-desktop__user-dropdown-button"
        data-testid="app-header-dropdown-trigger"
      >
        <div>
          <span
            data-testid="user-first-last-name"
          >
            Test
             
            Man
          </span>
        </div>
        <div
          class="icon header-desktop__dropdown-icon -xs-size"
          data-testid="icon"
        >
          <div />
        </div>
      </button>
    </div>
  </div>
</div>
`;

exports[`AppHeader should render correctly for teacher 1`] = `
<div>
  <div
    class="header-desktop -animation-open"
    data-testid="header-desktop"
  >
    <div
      class="header-desktop__user-dropdown-section"
      data-testid="dropdown-section"
    >
      <div
        class="avatar header-desktop__header-item"
        data-testid="avatar-image"
      >
        <div
          class="avatar__image sb-avatar"
          style="display: inline-block; vertical-align: middle; width: 32px; height: 32px; border-radius: 100%; font-family: Helvetica, Arial, sans-serif;"
        >
          <div
            class="avatar__image sb-avatar__text"
            style="width: 32px; height: 32px; line-height: initial; text-align: center; color: rgb(106, 110, 124); background: rgb(252, 252, 252); border-radius: 100%;"
            title="Test Man"
          >
            <div
              style="display: table; table-layout: fixed; width: 100%; height: 100%;"
            >
              <span
                style="display: table-cell; vertical-align: middle; font-size: 100%; white-space: nowrap;"
              >
                <span />
              </span>
            </div>
          </div>
        </div>
      </div>
      <button
        class="header-desktop__user-dropdown-button"
        data-testid="app-header-dropdown-trigger"
      >
        <div>
          <span
            data-testid="user-first-last-name"
          >
            Test
             
            Man
          </span>
        </div>
        <div
          class="icon header-desktop__dropdown-icon -xs-size"
          data-testid="icon"
        >
          <div />
        </div>
      </button>
    </div>
  </div>
</div>
`;
